test/unit/tedious-test.js

```diff
@@ -23,3 +23,27 @@ exports.connection = function(test) {
 
   test.done();
 };
+
+exports.connectionDoesNotModifyPassedConfig = function(test) {
+  var config = {
+    userName: 'sa',
+    password: 'sapwd',
+    options: {
+      port: 1234,
+      cryptoCredentialsDetails: {
+        ciphers: 'RC4-MD5'
+      }
+    }
+  };
+
+  var connection = new Connection(config);
+
+  test.notStrictEqual(connection.config, config);
+  test.notStrictEqual(connection.config.options, config.options);
+
+  // Test that we did not do a deep copy of the cryptoCredentialsDetails,
+  // as we never modify that value inside tedious.
+  test.strictEqual(connection.config.options.cryptoCredentialsDetails, config.options.cryptoCredentialsDetails);
```

Comment:
Do we want to pick and choose what we deep copy vs not based on implementation? It would seem like that'll introduce fragility in code.

And it's also not just about protecting the app from Tedious modifying, I think we also need to protect Tedious from the app modifying config.